Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift

```diff
@@ -143,6 +152,9 @@ extension BrowserViewController: TopToolbarDelegate {
 
   func topToolbarDidPressStop(_ topToolbar: TopToolbarView) {
     tabManager.selectedTab?.stop()
+    processAddressBarTask?.cancel()
+    topToolbarDidPressReloadTask?.cancel()
+    topToolbar.locationView.loading = tabManager.selectedTab?.loading ?? false
```

Comment:
When Cancel is pressed, shouldnt it be topToolbar.locationView.loading = false in any case ?

Reply:
Presumably yes that should be fine. Similar to above I am using the webview loading state to match current logic (webview loading state determine tab loading state) and avoid potential race conditions